src/App.tsx

```diff
@@ -1,56 +1,75 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
 import './App.scss';
+import cn from 'classnames';
 import { peopleFromServer } from './data/people';
+import { DropDownMenu } from './components/DropDownMenu/DropDownMenu';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [isSugActive, setIsSugActive] = useState(false);
+
+  const visiblePeople = useMemo(() => {
+    const normalizedQuery = appliedQuery.toLowerCase().trim();
+
+    setIsSugActive(!!normalizedQuery);
+
+    if (!normalizedQuery) {
+      setSelectedPerson(null);
+    }
+
+    return peopleFromServer.filter(person => (
+      person.name.toLowerCase().includes(normalizedQuery)
+    ));
+  }, [appliedQuery]);
+
+  const applyQuery = useCallback(
+    debounce(setAppliedQuery, 1000),
+    [],
+  );
+
+  const handleSelect = useCallback((
+    event: React.MouseEvent<HTMLAnchorElement, MouseEvent>,
+    person: Person,
+    status: boolean,
+  ) => {
+    event.preventDefault();
+    setSelectedPerson(person);
+    setQuery(person.name);
```

Comment:
You should clear appliedQuery when a person is selected. This will ensure that when the user starts typing again, the autocomplete will consider the new input, not the previously selected person's name.